src/test/scala/org/broadinstitute/dsde/firecloud/service/StatusServiceSpec.scala

```diff
@@ -0,0 +1,149 @@
+      status.intValue should be(InternalServerError.intValue)
+      val response = responseAs[SystemStatus]
+      response.ok should be(false)
+      response.systems("Agora").ok should be(false)
+      response.systems("Agora").messages should be(Some(List("Agora is down")))
+      response.systems.size should be(4)
+    }
+  }
+
+  agoraServer.when(
+    request().withMethod("GET").withPath("/status")
+  ).respond(
+    org.mockserver.model.HttpResponse.response()
+      .withHeaders(MockUtils.header)
+      .withBody("""{ "status": "up", "message": ["No problems with Agora"] }""")
+  )
+
+  "StatusService indicates OK" in {
+    Get("/status") ~> sealRoute(publicStatusRoutes) ~> check {
+      status.intValue should be(OK.intValue)
+      val response = responseAs[SystemStatus]
+      response.ok should be(true)
+      response.systems("Agora").ok should be(true)
+      response.systems("Agora").messages should be(None)
+      response.systems.size should be(4)
+    }
+  }
+
+  agoraServer.when(
+    request().withMethod("GET").withPath("/status")
+  ).respond(
+    org.mockserver.model.HttpResponse.response()
+      .withHeaders(MockUtils.header)
+      .withBody("bogus non-JSON response")
+  )
+
+  "StatusService indicates failure with non-JSON response" in {
+    Get("/status") ~> sealRoute(publicStatusRoutes) ~> check {
+      status.intValue should be(InternalServerError.intValue)
+      val response = responseAs[SystemStatus]
+      response.ok should be(false)
+      response.systems("Agora").ok should be(false)
+      response.systems("Agora").messages should be(Some(List("bogus non-JSON response")))
+      response.systems.size should be(4)
+    }
+  }
+
+  agoraServer.when(
+    request().withMethod("GET").withPath("/status")
+  ).respond(
+    org.mockserver.model.HttpResponse.response()
+      .withHeaders(MockUtils.header)
+      .withBody("")
+  )
+
+  "StatusService indicates failure with empty response" in {
+    Get("/status") ~> sealRoute(publicStatusRoutes) ~> check {
+      status.intValue should be(InternalServerError.intValue)
+      val response = responseAs[SystemStatus]
+      response.ok should be(false)
+      response.systems("Agora").ok should be(false)
+      response.systems("Agora").messages should be(Some(List("")))
+      response.systems.size should be(4)
+    }
+  }
+
+  agoraServer.when(
+    request().withMethod("GET").withPath("/status")
+  ).error((new HttpError).withDropConnection(true))
+
+  "StatusService reports failure when Agora drops connection" in {
+    Get("/status") ~> sealRoute(publicStatusRoutes) ~> check {
+      status.intValue should be(InternalServerError.intValue)
+      val response = responseAs[SystemStatus]
+      response.ok should be(false)
+      response.systems("Agora").ok should be(false)
+      response.systems("Agora").messages should be(Some(List("Premature connection close (the server doesn't appear to support request pipelining)")))
```

Comment:
I don't think it's worth testing the exact error message here, since it's out of our control. Maybe just check that messages is some non-empty list?